src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -197,55 +194,52 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
 
             CounterfactualLoggers.TransitiveDependencies.EmitIfNeeded(); // Emit only one event per VS session
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins = Enumerable.Empty<TransitivePackageReference>();
-            if (includeTransitivePackages)
+            if (includeTransitiveOrigins && await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token))
             {
-                if (includeTransitiveOrigins && await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token))
+                // Compute Transitive Origins
+                Dictionary<string, TransitiveEntry> transitiveOrigins;
+                if (includeTransitivePackages || IsInstalledAndTransitiveComputationNeeded
```

Comment:
This is the same condition as line 137 which had to be false for us to be in this else. Can you verify that we need to check both again and that the conditional logic is correct?